src/balloonpanel/balloonpanelview.js

```diff
@@ -123,263 +161,96 @@ export default class BalloonPanelView extends View {
 	}
 
 	/**
-	 * Attaches the balloon panel to a specified DOM element or range with a smart heuristics.
-	 *
-	 * **Notes**:
-	 *
-	 * * The algorithm takes the geometry of the "limiter element" into consideration so,
-	 * if possible, the balloon is positioned within the rect of that element.
-	 * * If possible, the balloon is positioned within the area of the "limiter element"
-	 * fitting into the browser viewport visible to the user. It prevents the panel from
-	 * appearing off screen.
-	 *
-	 * The heuristics chooses from among 4 available positions relative to the target DOM element or range:
-	 *
-	 * * South east:
-	 *
-	 *		[ Target ]
-	 *		    ^
-	 *		+-----------------+
-	 *		|                 |
-	 *		+-----------------+
-	 *
-	 *
-	 * * South west:
+	 * Attaches the balloon panel to a specified DOM element or range with a smart heuristics,
+	 * taking {@link ui.balloonPanel.BalloonPanelView#positions}, {@link ui.balloonPanel.BalloonPanelView#limiter}
+	 * and {@link ui.balloonPanel.BalloonPanelView#fitInViewport} into consideration.
 	 *
-	 *		         [ Target ]
-	 *		              ^
-	 *		+-----------------+
-	 *		|                 |
-	 *		+-----------------+
-	 *
-	 *
-	 * * North east:
-	 *
-	 *		+-----------------+
-	 *		|                 |
-	 *		+-----------------+
-	 *		    V
-	 *		[ Target ]
-	 *
-	 *
-	 * * North west:
-	 *
-	 *		+-----------------+
-	 *		|                 |
-	 *		+-----------------+
-	 *		              V
-	 *		         [ Target ]
-	 *
-	 * See {@ link ui.balloonPanel.BalloonPanelView#arrow}.
-	 *
-	 * @param {HTMLElement|Range} elementOrRange Target DOM element or range to which the balloon will be attached.
-	 * @param {HTMLElement|Object} limiterElementOrRect The DOM element or element rect beyond which area the balloon panel should not be
-	 * positioned, if possible.
+	 * @param {[type]} target [description]
+	 * @returns {[type]} [description]
 	 */
-	attachTo( elementOrRange, limiterElementOrRect ) {
```

Comment:
It depends on whether you'll be changing object to which the balloon is attached and the limiter. I think they go in pairs, so setting both at once makes more sense to me. If you'll decide to leave it separately then on the  limiter change the balloon should reposition it self, of course. But that would also leave it unclear why the anchor is set through `attachTo()` and limitter through a setter.